### DISCUSSION TITLE
Use Plausible analytics offline with workbox

### DISCUSSION THREAD

**Discussion by zusorio** [plausible/analytics#1424] [Anything Else]
I recently migrated from using google analytics to Plausible for my PWA which uses workbox. I was using the workbox-google-analytics plugin to store analytics while offline and then send them afterwards, how would I achieve something similar with Plausible?

**Comment by metmarkosaric** [comment]
hi @ZusorCode! i'm not familiar with workbox unfortunately and nobody yet has created a workbox-plausible plugin from what i'm aware of. we don't have any offline functionality by default in our script but i do know that [this plugin](https://github.com/ian/analytics/tree/main/packages/plausible) advertises as adding offline functionality to Plausible so perhaps you can try it?

**Reply by zusorio** [reply]
Is there any way to correct the time that analytics were tracked at (send analytics later with an earlier date)? With that I don't think it'd be that hard to make my own plugin for workbox.

**Reply by ukutaht** [reply]
Unfortunately there's no option to send a timestamp from the client-side at the moment.

I'm curious, when you send offline events after the fact to GA, does session tracking work as expected?

Plausible keeps live sessions in memory for up to 30 minutes after the session is over. So if we decided to support something like this and you send events more than 30 minutes after they happened, we wouldn't be able to link it to an existing session.
